orbit/pkg/platform/platform_windows.go

```diff
@@ -341,3 +344,203 @@ func GetSMBiosUUID() (string, UUIDSource, error) {
+	if len(outputByLines) < 1 {
+		return "", errors.New("expected number of lines is not present")
+	}
+
+	rawVersionStr := strings.TrimSpace(strings.ToLower(outputByLines[0]))
+	if !strings.HasPrefix(rawVersionStr, expectedPrefix) {
+		return "", errors.New("expected version prefix is not present")
+	}
+
+	// getting the actual version string
+	versionStr := strings.ReplaceAll(rawVersionStr, expectedPrefix, "")
+	if len(versionStr) == 0 {
+		return "", errors.New("expected version information is not present")
+	}
+
+	return versionStr, nil
+}
+
+// fixSymlinkNotPresent fixes the issue where the symlink to the orbit service binary is not present
+// this is a workaround for the issue described here https://github.com/fleetdm/fleet/issues/10300
+func fixSymlinkNotPresent() error {
+	// getting current working directory
+	execPath, err := getWorkingPath()
+	if err != nil {
+		return err
+	}
+
+	// getting the path to orbit service binary
+	orbitPath := execPath + "\\..\\bin\\orbit\\orbit.exe"
+
+	// gathering target orbit version
+	versionOrbit, err := getOrbitVersion(orbitPath)
+	if err != nil {
+		return fmt.Errorf("getting orbit version: %w", err)
+	}
+
+	// checking if target orbit has the problematic logic
+	if versionOrbit != "1.6.0" && versionOrbit != "1.7.0" {
+		return nil
+	}
+
+	// checking if the orbit service binary symlink needs to be regenerated
+	_, err = os.Readlink(orbitPath)
+
+	// if there are no errors, it means that the symlink is present, nothing to do
+	if err == nil {
+		return nil
+	}
+
+	// handling error by renaming the locked binary file, marking it for deletion on reboot and
+	// regenerating the symlink
+	if !errors.Is(err, os.ErrNotExist) {
+
+		// symlink is going to be recreated - so better check that symlink target is present
+		targetSymlinkPath := ""
+		stableOrbitPath := execPath + "\\..\\bin\\orbit\\windows\\stable\\orbit.exe"
+		edgeOrbitPath := execPath + "\\..\\bin\\orbit\\windows\\edge\\orbit.exe"
+
+		if _, err := os.Stat(stableOrbitPath); err == nil {
+			targetSymlinkPath = stableOrbitPath
+		} else if _, err := os.Stat(edgeOrbitPath); err == nil {
+			targetSymlinkPath = edgeOrbitPath
+		}
+
+		if len(targetSymlinkPath) == 0 {
+			return errors.New("symlink target cannot be found")
+		}
+
+		// We are now about to perform a sensitive operation
+		// by first renaming current orbit file and then recreating recreating the symlink
+
+		// renaming locked binary to a different file, the process will keep running, but it will be renamed
+		// target process is not terminated on purpose to avoid potential erros
+		targetOrbitPath := orbitPath + "." + strings.ToUpper(uuid.New().String())
+		if err := os.Rename(orbitPath, targetOrbitPath); err != nil {
+			return fmt.Errorf("rename: %w", err)
+		}
+
+		// regenerating the symlink
+		if err := os.Symlink(targetSymlinkPath, orbitPath); err != nil {
+			return fmt.Errorf("symlink current: %w", err)
```

Comment:
Same question here. Should we do any checks if we failed?

Reply:
not much we can do I'm afraid as any other action might introduce risks